test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -0,0 +1,345 @@
+		await expect(
+			page.getByRole( 'document', {
+				name: 'Block: Heading',
+			} )
+		).toBeFocused();
+	} );
+
+	/**
+	 * When all the blocks gets removed from the editor, it inserts a default
+	 * paragraph block; make sure that paragraph block gets selected after
+	 * removing blocks from ListView.
+	 */
+	test( 'selects the default paragraph block after removing all blocks', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Select the Image block as well.
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowUp' );
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Image link',
+				selected: true,
+			} )
+			.waitFor();
+
+		// Remove both blocks.
+		await page
+			.getByRole( 'button', { name: 'Options for Image block' } )
+			.click();
+		await page.getByRole( 'menuitem', { name: /Remove blocks/i } ).click();
+
+		// Newly created paragraph block should be selected.
+		await expect(
+			page.getByRole( 'document', { name: /Empty block/i } )
+		).toBeFocused();
+	} );
+
+	test( 'expands nested list items', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Click first color option from the block placeholder's color picker to
+		// make the inner blocks appear.
+		await page
+			.getByRole( 'button', { name: /Color: /i } )
+			.first()
+			.click();
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Things start off expanded.
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Cover link',
+				expanded: true,
+			} )
+			.waitFor();
+
+		// The child paragraph block should be selected.
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Paragraph link',
+				selected: true,
+			} )
+			.waitFor();
+
+		// Collapse the Cover block.
+		await page
+			.getByRole( 'gridcell', { name: 'Cover link' } )
+			.locator( '.block-editor-list-view__expander[aria-hidden="true"]' )
+			// Force the click to bypass the visibility check. The expander is
+			// intentionally aria-hidden. See the implementation for details.
+			.click( { force: true } );
+
+		// Check that we're collapsed.
+		await expect(
+			page.getByRole( 'gridcell', { name: /link/i } )
+		).toHaveCount( 1 );
+
+		// Click the Cover block List View item.
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Cover link',
+				expanded: false,
+			} )
+			.click();
+
+		// Click the Cover block title placeholder.
+		await page
+			.getByRole( 'document', { name: 'Block: Cover' } )
+			.getByRole( 'document', { name: /Empty block/i } )
+			.click();
+
+		// The child paragraph block in List View should be selected.
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Paragraph link',
+				selected: true,
+			} )
+			.waitFor();
```

Comment:
Let's do `.toBeVisible()` or other explicit assertions here. It's not ideal compared to something like `.toBeSelected()` but Playwright doesn't seem to support it at the time.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/47919/commits/2193b792b6ea60076682b8b34a292f7783181c4c.